src/main/java/org/broadinstitute/dsm/jobs/UPSTrackingJob.java

```diff
@@ -164,65 +165,59 @@ private static void updateTrackingInfo(String statusType,
-            SimpleResult dbVals = new SimpleResult();
-            try (PreparedStatement stmt = conn.prepareStatement(query)) {
-                stmt.setString(1, upsUpdate);
-                stmt.setString(2, date);
-                stmt.setString(3, trackingId);
-                int r = stmt.executeUpdate();
-                if (r != 2) {//number of subkits
-                    throw new RuntimeException("Update query for UPS tracking updated " + r + " rows! with tracking/return id: " + trackingId);
+        if (!isReturn) {
+            if (shouldTriggerEventForKitOnItsWayToParticipant(statusType, oldType)) {
+                KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] {DELIVERED, trackingId}, 2);//todo change this to the number of subkits but for now 2 for test boston works
+                if (kitDDPNotification != null) {
+                    logger.info("Triggering DDP for kit going to participant with external order number: " + kit.getExternalOrderNumber());
+                    EventUtil.triggerDDP(kitDDPNotification);
+                } else {
+                    logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
                 }
             }
-            catch (Exception e) {
-                dbVals.resultException = e;
-            }
-            return dbVals;
-        });
-        if (result.resultException != null) {
-            throw new RuntimeException(result.resultException);
-        }
-        else {
-            if (!isReturn) {
-                if (shouldTriggerEventForKitOnItsWayToParticipant(statusType, oldType)) {
-                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
-                    if (kitDDPNotification != null) {
-                        logger.info("Triggering DDP for kit going to participant with external order number: " + kit.getExternalOrderNumber());
-                        EventUtil.triggerDDP(kitDDPNotification);
-                    }
-                    else {
-                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
-                    }
+        } else { // this is a return
+            if (earliestInTransitTime != null && !kit.isCEOrdered()) {
+                // if we have the first date of an inbound event, create an order in CE
+                // using the earliest date of inbound event
+                OrderResponse orderResponse = orderRegistrar.orderTest(DSMServer.careEvolveAuth, kit.getHRUID(), kit.getKitLabel(), kit.getExternalOrderNumber(), earliestInTransitTime);
```

Comment:
While here, I figured we should check the `orderResponse` in case there's an error.